diff --git a/dom/reporting/ReportingHeader.cpp b/dom/reporting/ReportingHeader.cpp
--- a/dom/reporting/ReportingHeader.cpp
+++ b/dom/reporting/ReportingHeader.cpp
@@ -224,7 +224,8 @@
 /* static */
 EndpointsList ReportingHeader::ProcessReportingEndpointsListFromResponse(
     nsIHttpChannel* aChannel) {
-  if (!StaticPrefs::dom_reporting_enabled()) {
+  if (!StaticPrefs::dom_reporting_enabled() ||
+      !StaticPrefs::dom_reporting_header_enabled()) {
     return {};
   }
 
