--- a/browser/components/preferences/main.js
+++ b/browser/components/preferences/main.js
@@ -1517,208 +1517,8 @@
   visible: ({ enableDeletePrivate }) => enableDeletePrivate.value,
   onUserChange() {
     Services.prefs.setBoolPref("browser.download.deletePrivate.chosen", true);
-  },
-});
-/**
- * A helper object containing all logic related to
- * setting the browser as the user's default.
- */
-const DefaultBrowserHelper = {
-  /**
-   * @type {number}
-   */
-  _backoffIndex: 0,
-
-  /**
-   * @type {number | undefined}
-   */
-  _pollingTimer: undefined,
-
-  /**
-   * Keeps track of the last known browser
-   * default value set to compare while polling.
-   *
-   * @type {boolean | undefined}
-   */
-  _lastPolledIsDefault: undefined,
-
-  /**
-   * @type {typeof import('../shell/ShellService.sys.mjs').ShellService | undefined}
-   */
-  get shellSvc() {
-    return (
-      AppConstants.HAVE_SHELL_SERVICE &&
-      // @ts-ignore from utilityOverlay.js
-      getShellService()
-    );
-  },
-
-  /**
-   * Sets up polling of whether the browser is set to default,
-   * and calls provided hasChanged function when the state changes.
-   *
-   * @param {Function} hasChanged
-   */
-  pollForDefaultChanges(hasChanged) {
-    if (this._pollingTimer) {
-      return;
-    }
-    this._lastPolledIsDefault = this.isBrowserDefault;
-
-    // Exponential backoff mechanism will delay the polling times if user doesn't
-    // trigger SetDefaultBrowser for a long time.
-    const backoffTimes = [
-      1000, 1000, 1000, 1000, 2000, 2000, 2000, 5000, 5000, 10000,
-    ];
-
-    const pollForDefaultBrowser = () => {
-      if (
-        (location.hash == "" ||
-          location.hash == "#general" ||
-          location.hash == "#sync") &&
-        document.visibilityState == "visible"
-      ) {
-        const { isBrowserDefault } = this;
-        if (isBrowserDefault !== this._lastPolledIsDefault) {
-          this._lastPolledIsDefault = isBrowserDefault;
-          hasChanged();
-        }
-      }
-
-      if (!this._pollingTimer) {
-        return;
-      }
-
-      // approximately a "requestIdleInterval"
-      this._pollingTimer = window.setTimeout(
-        () => {
-          window.requestIdleCallback(pollForDefaultBrowser);
-        },
-        backoffTimes[
-          this._backoffIndex + 1 < backoffTimes.length
-            ? this._backoffIndex++
-            : backoffTimes.length - 1
-        ]
-      );
-    };
-
-    this._pollingTimer = window.setTimeout(() => {
-      window.requestIdleCallback(pollForDefaultBrowser);
-    }, backoffTimes[this._backoffIndex]);
-  },
-
-  /**
-   * Stops timer for polling changes.
-   */
-  clearPollingForDefaultChanges() {
-    if (this._pollingTimer) {
-      clearTimeout(this._pollingTimer);
-      this._pollingTimer = undefined;
-    }
-  },
-
-  /**
-   *  Checks if the browser is default through the shell service.
-   */
-  get isBrowserDefault() {
-    if (!this.canCheck) {
-      return false;
-    }
-    return this.shellSvc?.isDefaultBrowser(false, true);
-  },
-
-  /**
-   * Attempts to set the browser as the user's
-   * default through the shell service.
-   *
-   * @returns {Promise<void>}
-   */
-  async setDefaultBrowser() {
-    // Reset exponential backoff delay time in order to do visual update in pollForDefaultBrowser.
-    this._backoffIndex = 0;
-
-    try {
-      await this.shellSvc?.setDefaultBrowser(false);
-    } catch (e) {
-      console.error(e);
-    }
-  },
-
-  /**
-   * Checks whether the browser is capable of being made default.
-   *
-   * @type {boolean}
-   */
-  get canCheck() {
-    return (
-      this.shellSvc &&
-      /**
-       * Flatpak does not support setting nor detection of default browser
-       */
-      !gGIOService?.isRunningUnderFlatpak
-    );
   },
-};
-
-Preferences.addSetting({
-  id: "alwaysCheckDefault",
-  pref: "browser.shell.checkDefaultBrowser",
-  setup: emitChange => {
-    if (!DefaultBrowserHelper.canCheck) {
-      return;
-    }
-    DefaultBrowserHelper.pollForDefaultChanges(emitChange);
-    // eslint-disable-next-line consistent-return
-    return () => DefaultBrowserHelper.clearPollingForDefaultChanges();
-  },
-  /**
-   * Show button for setting browser as default browser or information that
-   * browser is already the default browser.
-   */
-  visible: () => DefaultBrowserHelper.canCheck,
-  disabled: (_, setting) =>
-    !DefaultBrowserHelper.canCheck ||
-    setting.locked ||
-    DefaultBrowserHelper.isBrowserDefault,
 });
-
-Preferences.addSetting({
-  id: "isDefaultPane",
-  deps: ["alwaysCheckDefault"],
-  visible: () =>
-    DefaultBrowserHelper.canCheck && DefaultBrowserHelper.isBrowserDefault,
-});
-
-Preferences.addSetting({
-  id: "isNotDefaultPane",
-  deps: ["alwaysCheckDefault"],
-  visible: () =>
-    DefaultBrowserHelper.canCheck && !DefaultBrowserHelper.isBrowserDefault,
-  onUserClick: (e, { alwaysCheckDefault }) => {
-    if (!DefaultBrowserHelper.canCheck) {
-      return;
-    }
-    const setDefaultButton = /** @type {MozButton} */ (e.target);
-
-    if (!setDefaultButton) {
-      return;
-    }
-    if (setDefaultButton.disabled) {
-      return;
-    }
-
-    /**
-     * Disable the set default button, so that the user
-     * doesn't try to hit it again while browser is being set to default.
-     */
-    setDefaultButton.disabled = true;
-    alwaysCheckDefault.value = true;
-    DefaultBrowserHelper.setDefaultBrowser().finally(() => {
-      setDefaultButton.disabled = false;
-    });
-  },
-});
-
 // Firefox support settings
 Preferences.addSetting({
   id: "supportLinksGroup",
@@ -2606,47 +2406,6 @@
     }
   }
 );
-
-function createDefaultBrowserConfig({
-  includeIsDefaultPane = true,
-  inProgress = false,
-  hiddenFromSearch = false,
-} = {}) {
-  const isDefaultPane = {
-    id: "isDefaultPane",
-    l10nId: "is-default-browser-2",
-    control: "moz-promo",
-  };
-
-  const isNotDefaultPane = {
-    id: "isNotDefaultPane",
-    l10nId: "is-not-default-browser-2",
-    control: "moz-promo",
-    options: [
-      {
-        control: "moz-button",
-        l10nId: "set-as-my-default-browser-2",
-        id: "setDefaultButton",
-        slot: "actions",
-        controlAttrs: {
-          type: "primary",
-        },
-      },
-    ],
-  };
-
-  const items = includeIsDefaultPane
-    ? [isDefaultPane, isNotDefaultPane]
-    : [isNotDefaultPane];
-
-  return {
-    l10nId: "home-default-browser-title",
-    headingLevel: 2,
-    items,
-    ...(inProgress && { inProgress }),
-    ...(hiddenFromSearch && { hiddenFromSearch }),
-  };
-}
 
 function createStartupConfig(hidden = false) {
   return {
@@ -2687,10 +2446,6 @@
         id: "windowsLaunchOnLoginDisabledProfileBox",
         control: "moz-message-bar",
         l10nId: "startup-windows-launch-on-login-profile-disabled",
-      },
-      {
-        id: "alwaysCheckDefault",
-        l10nId: "always-check-default",
       },
     ],
   };
@@ -2753,7 +2508,6 @@
       },
     ],
   },
-  defaultBrowser: createDefaultBrowserConfig(),
   startup: createStartupConfig(
     Services.prefs.getBoolPref("browser.settings-redesign.enabled", false)
   ),
@@ -4627,11 +4381,6 @@
       },
     ],
   },
-  defaultBrowserSync: createDefaultBrowserConfig({
-    includeIsDefaultPane: false,
-    inProgress: true,
-    hiddenFromSearch: true,
-  }),
   sync: {
     inProgress: true,
     l10nId: "sync-group-label",
@@ -5215,7 +4964,6 @@
     initSettingGroup("translations");
     initSettingGroup("spellCheck");
     initSettingGroup("performance");
-    initSettingGroup("defaultBrowser");
     initSettingGroup("startup");
     initSettingGroup("importBrowserData");
     initSettingGroup("networkProxy");
